### ISSUE TITLE
Data Normalization Error?

### ISSUE THREAD

**Issue by djaroslawicz** [baccuslab/deep-retina#13]
Hello,

In experiments.py (line 78/80) you normalize the experiment data immediately upon loading it. Shouldn't you first split the data into train and validation sets before calculating normalization statistics?

Best,
Daniel

**Comment by lmcintosh** [comment]
Hi Daniel,

Note that we are normalizing the stimulus data stored as
"f[train_or_test]['stimulus']", where the variable train_or_test is either
'train' or 'test'. So our normalization during training is independent of
the normalization during test time.

Best,
Lane


--
Lane McIntosh
www.lanemcintosh.com

**Comment by djaroslawicz** [comment]
Thanks for the response. Isn't this still problematic for the validation data set, which is only produced in core.py when you call mdl.fit?

**Comment by lmcintosh** [comment]
This normalization is simply subtracting the mean stimulus value and dividing by the standard deviation of the stimulus. If you have a very small validation data set then it is possible that your train and validation mean, std values are different, but since we are generating our stimulus from the same dataset of natural images and have large validation splits, it seems unlikely to me that normalizing independently or dependently for the validation dataset will make any difference in the hyperparameters we select using the validation data.
